Sources/Brave/Frontend/Browser/Helpers/LaunchHelper.swift

```diff
@@ -127,6 +123,27 @@ public actor LaunchHelper {
   }
 }
 
+private extension FilterListStorage {
+  /// Return all the blocklist types that are valid for filter lists.
+  var validBlocklistTypes: Set<ContentBlockerManager.BlocklistType> {
+    if filterLists.isEmpty {
+      /// If we don't have filter lists yet loaded, use the settings
+      return allFilterListSettings.reduce(Set<ContentBlockerManager.BlocklistType>()) { partialResult, setting in
+        guard let componentId = setting.componentId else { return partialResult }
+        return partialResult.union([
+          .filterList(componentId: componentId, isAlwaysAggressive: setting.isAlwaysAggressive)
+        ])
+      }
+    } else {
+      /// If we do have filter lists yet loaded, use them as they are always the most up to date and accurate
+      return filterLists.reduce(Set<ContentBlockerManager.BlocklistType>()) { partialResult, filterList in
+        return partialResult.union([
+          .filterList(componentId: filterList.entry.componentId, isAlwaysAggressive: filterList.isAlwaysAggressive)
+        ])
```

Comment:
Use mutable variant of reduce like above and just call `insert` here, no need to form a union with a single item